pages/mesosphere/dcos/services/include/security-create-permissions.tmpl

```diff
@@ -1,89 +1,38 @@
 
 # <a name="give-perms"></a>Create and Assign Permissions
-Use the following `curl` commands to rapidly provision the {{ model.techName }} service account with the required permissions.
-
-<p class="message--note"><strong>NOTE: </strong>Any forward slash ("/") in a resource must be replaced with `%252F` before it can be passed in a curl command.</p>
-
-When using the API to manage permissions, you must first create the permissions and then assign them. Sometimes, the permission may already exist. In this case, the API returns an informative message. You can regard this as a confirmation and continue to the next command.
+Use the following DC/OS CLI commands to rapidly provision the {{ model.techName }} service account with the required permissions.
 
 1.  Create the permission.
 
-<p class="message--important"><strong>IMPORTANT: </strong>These commands use the default {{ model.techName }} <tt>role</tt> value of <tt>{{ model.packageName }}-role</tt>. If you are running multiple instances of {{ model.techName }}, replace the instances of <tt>{{ model.packageName }}-role</tt> with the correct name (<tt>&lt;name&gt;-role</tt>). For example, if you have a {{ model.techName }} instance named <tt>{{ model.packageName }}2</tt>, you would replace each role value in the code samples to <tt>{{ model.packageName }}2-role</tt>.</p>
+<p class="message--important"><strong>IMPORTANT: </strong>The value to be used for <tt>&lt;service-role&gt;</tt> will be based on the service name. The table below shows a few examples to show the pattern used by the service as it Mesos role to use. If you need help configuring the permissions for {{ model.packageName }}, please feel to reach out to D2iQ support by filing a support ticket. Replace the instances of <tt>&lt;service-role&gt;</tt> with the correct name (<tt>&lt;name&gt;-role</tt>).</p>
```

Comment:
@surdy this needs to be changed for DC/OS 2.0 for multi-tenancy based on whether `enforceRole=true` is set on the marathon group.

I would split the `service-role` column into two, one where `enforceRole=true` and `enforceRole=false`
```
| Service name                           |                                      &lt;service-role&gt;                                       |
|                                        |            enforceRole=false                   |            enforceRole=true                    |
|----------------------------------------|------------------------------------------------|------------------------------------------------|
| `/{{ model.packageName }}`             | `{{ model.packageName }}-role`                 | `{{ model.packageName }}-role`                 |
| `/{{ model.packageName }}-prod`        | `{{ model.packageName }}-prod-role`            | `{{ model.packageName }}-prod-role`            |
| `/team01/{{ model.packageName }}`      | `team01__{{ model.packageName }}-role`         | `team01`                                       |
| `/team01/prod/{{ model.packageName }}` | `team01__prod__{{ model.packageName }}-role`   | `team01`                                       |
```

Additionally, for SDK based services, if the user intends to *migrate* existing running payloads from the legacy role to the new quota enforced roles, they need to provide security permissions for *both* roles.

eg: for an SDK based service running under `/team01/prod/{{ model.packageName }}` which upgrades to a quota aware one and the user wishes to move pods to the quota role `team01` the user will need to provide permissions for these two:
- `team01__prod__{{ model.packageName }}-role` (legacy-role)
- `team01` (quota-enforced-role)

This is outlined in the [SDK 0.57.0 Release Notes](https://github.com/mesosphere/dcos-commons/releases/tag/0.57.0) under `Quota Support`.

Happy to answer questions as this can be a bit confusing.

Reply:
Thanks @kaiwalyajoshi. I am familiar with that change in DC/OS 2.0 and 0.57 sdk. I had that as part of this commit but took it out as it seemed like two much for one or. I will update it this one with my previous changes